environs/config/config.go

```diff
@@ -990,6 +1011,20 @@ func (c *Config) CAPrivateKey() (key string, ok bool) {
 	return "", false
 }
 
+// RsyslogConfig return the rsyslog client config.
+func (c *Config) RsyslogConfig() (*rsyslog.ClientConfig, error) {
```

Comment:
huh, yeah, we could just use pointers instead of worrying about zero values, but that doesn't quite sit right with me. I particularly like Configgy things to be values to minimise (if not eliminate) the risks of sharing reference across components.